File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        res = []
        
        num_visited = set()
        for n in nums:
            if n in num_visited:
                res.append(n)
            else:
                num_visited.add(n)
        
        return res

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        unordered_set<int> hashset;
        vector<int> res;

        for (int i=0; i < nums.size(); i ++) {
            if ( hashset.contains(nums[i])) {
                res.push_back(nums[i]);
            }
            else {
                hashset.insert(nums[i]);
            }
        }

        return res;


        
    }
};